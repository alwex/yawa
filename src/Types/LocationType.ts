enum LocationType {
  street = 'street',
  subcity = 'subcity',
  city = 'city',
  county = 'county',
  state = 'state',
}

export default LocationType
