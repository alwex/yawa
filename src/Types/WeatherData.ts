interface WeatherData {
  dt: number
  main: {
    temp: number
    temp_min: number
    temp_max: number
    pressure: number
    sea_level: number
    grnd_level: number
    humidity: number
    temp_kf: number
  }
  weather: {
    id: number
    main: string
    description: string
    icon: string
  }
  clouds: {
    all: number
  }
  wind: {
    speed: number
    deg: number
  }
  rain: {}
  sys: {
    pod: string
  }
  dt_txt: string
}

export default WeatherData

const sample = {
  dt: 1545458400,
  main: {
    temp: 293.08,
    temp_min: 293.079,
    temp_max: 293.08,
    pressure: 1010.25,
    sea_level: 1020.63,
    grnd_level: 1010.25,
    humidity: 60,
    temp_kf: 0,
  },
  weather: [
    {
      id: 804,
      main: 'Clouds',
      description: 'overcast clouds',
      icon: '04d',
    },
  ],
  clouds: {
    all: 88,
  },
  wind: {
    speed: 2.97,
    deg: 270.008,
  },
  rain: {},
  sys: {
    pod: 'd',
  },
  dt_txt: '2018-12-22 06:00:00',
}
