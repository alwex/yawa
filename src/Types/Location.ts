import LocationType from './LocationType'

interface Location {
  alternative_names: string
  boundingbox: [number, number, number, number]
  display_name: string
  lon: number
  lat: number
  name: string
  name_suffix: string
  type: LocationType
}

export default Location
