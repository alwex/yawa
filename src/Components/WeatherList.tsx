import React from 'react'
import WeatherData from '../Types/WeatherData'
import { List, ListItem } from 'native-base'
import WeatherItem from './WeatherItem'
import Location from '../Types/Location'

interface WeatherListProps {
  onPress: (forecast: WeatherData) => void
  forecasts: WeatherData[]
  location: Location
}

export default class WeatherList extends React.Component<WeatherListProps> {
  render() {
    const { forecasts, location, onPress } = this.props
    return (
      <List>
        {forecasts.map((forecast: WeatherData) => (
          <ListItem key={forecast.dt}>
            <WeatherItem
              onPress={() => onPress(forecast)}
              date={forecast.dt_txt}
              temp={parseInt(forecast.main.temp.toFixed(0))}
              humidity={forecast.main.humidity}
              description={forecast.weather[0].description}
              iconCode={forecast.weather[0].icon}
              locationName={location.name}
            />
          </ListItem>
        ))}
      </List>
    )
  }
}
