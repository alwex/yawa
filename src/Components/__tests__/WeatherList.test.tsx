import React from 'react'
import renderer from 'react-test-renderer'
import WeatherList from '../WeatherList'
import Location from '../../Types/Location'
import LocationType from '../../Types/LocationType'
import WeatherData from '../../Types/WeatherData'

describe('Loader', () => {
  it('render correctly', () => {
    const location1: Location = {
      alternative_names: 'Dunedin',
      boundingbox: [-77.4369278, 37.2065392, -77.4369278, 37.2065392],
      display_name: 'Dunedin, Petersburg, Virginia',
      lon: -77.4369278,
      lat: 37.2065392,
      name: 'Dunedin',
      name_suffix: 'Petersburg, Virginia',
      type: LocationType.city,
    }

    const forecast1 = {
      dt: 1545609600,
      main: {
        temp: 20.06,
        temp_min: 20.06,
        temp_max: 20.06,
        pressure: 990.72,
        sea_level: 1030.52,
        grnd_level: 990.72,
        humidity: 73,
        temp_kf: 0,
      },
      weather: [
        {
          id: 800,
          main: 'Clear',
          description: 'clear sky',
          icon: '01d',
        },
      ],
      clouds: {
        all: 0,
      },
      wind: {
        speed: 1.59,
        deg: 67.5001,
      },
      sys: {
        pod: 'd',
      },
      dt_txt: '2018-12-24 00:00:00',
    }

    const forecast2 = {
      dt: 1545620400,
      main: {
        temp: 20.84,
        temp_min: 20.84,
        temp_max: 20.84,
        pressure: 990.42,
        sea_level: 1030.05,
        grnd_level: 990.42,
        humidity: 72,
        temp_kf: 0,
      },
      weather: [
        {
          id: 800,
          main: 'Clear',
          description: 'clear sky',
          icon: '02d',
        },
      ],
      clouds: {
        all: 8,
      },
      wind: {
        speed: 1.83,
        deg: 93.0087,
      },
      sys: {
        pod: 'd',
      },
      dt_txt: '2018-12-24 03:00:00',
    }

    const weatherData: WeatherData[] = [forecast1 as WeatherData, forecast2 as WeatherData]

    const component = renderer
      .create(<WeatherList onPress={() => {}} location={location1} forecasts={weatherData} />)
      .toJSON()
    expect(component).toMatchSnapshot()
  })
})
