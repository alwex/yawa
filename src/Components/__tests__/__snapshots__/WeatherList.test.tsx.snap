// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader render correctly 1`] = `
<View
  closeOnRowBeginSwipe={false}
  closeOnRowPress={true}
  closeOnScroll={true}
  directionalDistanceChangeThreshold={2}
  disableLeftSwipe={false}
  disableRightSwipe={false}
  leftOpenValue={0}
  previewFirstRow={false}
  recalculateHiddenLayout={false}
  rightOpenValue={0}
  style={Object {}}
  swipeToOpenPercent={50}
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 0.5,
          "borderColor": "#c9c9c9",
          "flexDirection": "row",
          "marginLeft": 16,
          "paddingBottom": 13,
          "paddingRight": 16,
          "paddingTop": 13,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-around",
              },
            ]
          }
        >
          <View
            style={Object {}}
          >
            <View
              style={Object {}}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#2C3A3F",
                      "fontSize": 50,
                    },
                  ]
                }
                uppercase={false}
              >
                20
                °
              </Text>
            </View>
            <View
              style={Object {}}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#D6637A",
                      "fontSize": 12,
                      "fontWeight": "bold",
                    },
                  ]
                }
                uppercase={false}
              >
                CLEAR SKY
              </Text>
            </View>
            <View
              style={Object {}}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#90C7D4",
                      "fontSize": 12,
                      "fontWeight": "bold",
                    },
                  ]
                }
                uppercase={false}
              >
                DUNEDIN
              </Text>
            </View>
          </View>
          <View
            style={Object {}}
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#D6637A",
                      "fontSize": 80,
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              type="MaterialCommunityIcons"
            >
              
            </Text>
          </View>
          <View
            style={Object {}}
          >
            <View
              style={Object {}}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignSelf": "flex-start",
                      "backgroundColor": "#ED1727",
                      "borderRadius": 13.5,
                      "height": 27,
                      "justifyContent": "center",
                      "paddingBottom": 3,
                      "paddingLeft": 6,
                      "paddingRight": 6,
                      "paddingTop": 3,
                    },
                    Object {
                      "alignSelf": "center",
                      "backgroundColor": "#D6637A",
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#000",
                          "fontSize": 30,
                        },
                        Object {
                          "color": "#000",
                          "fontSize": 30,
                        },
                        Object {
                          "alignSelf": "center",
                          "color": "white",
                          "fontSize": 14,
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  type="MaterialCommunityIcons"
                >
                  
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#fff",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "lineHeight": 19,
                      "paddingLeft": 3,
                      "paddingRight": 3,
                      "textAlign": "center",
                    }
                  }
                  uppercase={false}
                >
                  73
                  %
                </Text>
              </View>
            </View>
            <View
              style={Object {}}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    Object {
                      "alignSelf": "center",
                      "color": "#C9E4EB",
                      "fontSize": 50,
                      "fontWeight": "bold",
                    },
                  ]
                }
                uppercase={false}
              >
                24
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    Object {
                      "alignSelf": "center",
                      "color": "#90C7D4",
                      "fontSize": 30,
                      "fontWeight": "bold",
                    },
                  ]
                }
                uppercase={false}
              >
                MON
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    Object {
                      "alignSelf": "center",
                      "color": "#2C3A3F",
                      "fontSize": 12,
                      "fontWeight": "normal",
                    },
                  ]
                }
                uppercase={false}
              >
                1:00 PM
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 0.5,
          "borderColor": "#c9c9c9",
          "flexDirection": "row",
          "marginLeft": 16,
          "paddingBottom": 13,
          "paddingRight": 16,
          "paddingTop": 13,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-around",
              },
            ]
          }
        >
          <View
            style={Object {}}
          >
            <View
              style={Object {}}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#2C3A3F",
                      "fontSize": 50,
                    },
                  ]
                }
                uppercase={false}
              >
                21
                °
              </Text>
            </View>
            <View
              style={Object {}}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#D6637A",
                      "fontSize": 12,
                      "fontWeight": "bold",
                    },
                  ]
                }
                uppercase={false}
              >
                CLEAR SKY
              </Text>
            </View>
            <View
              style={Object {}}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#90C7D4",
                      "fontSize": 12,
                      "fontWeight": "bold",
                    },
                  ]
                }
                uppercase={false}
              >
                DUNEDIN
              </Text>
            </View>
          </View>
          <View
            style={Object {}}
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#D6637A",
                      "fontSize": 80,
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              type="MaterialCommunityIcons"
            >
              
            </Text>
          </View>
          <View
            style={Object {}}
          >
            <View
              style={Object {}}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignSelf": "flex-start",
                      "backgroundColor": "#ED1727",
                      "borderRadius": 13.5,
                      "height": 27,
                      "justifyContent": "center",
                      "paddingBottom": 3,
                      "paddingLeft": 6,
                      "paddingRight": 6,
                      "paddingTop": 3,
                    },
                    Object {
                      "alignSelf": "center",
                      "backgroundColor": "#D6637A",
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#000",
                          "fontSize": 30,
                        },
                        Object {
                          "color": "#000",
                          "fontSize": 30,
                        },
                        Object {
                          "alignSelf": "center",
                          "color": "white",
                          "fontSize": 14,
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  type="MaterialCommunityIcons"
                >
                  
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#fff",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "lineHeight": 19,
                      "paddingLeft": 3,
                      "paddingRight": 3,
                      "textAlign": "center",
                    }
                  }
                  uppercase={false}
                >
                  72
                  %
                </Text>
              </View>
            </View>
            <View
              style={Object {}}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    Object {
                      "alignSelf": "center",
                      "color": "#C9E4EB",
                      "fontSize": 50,
                      "fontWeight": "bold",
                    },
                  ]
                }
                uppercase={false}
              >
                24
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    Object {
                      "alignSelf": "center",
                      "color": "#90C7D4",
                      "fontSize": 30,
                      "fontWeight": "bold",
                    },
                  ]
                }
                uppercase={false}
              >
                MON
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    Object {
                      "alignSelf": "center",
                      "color": "#2C3A3F",
                      "fontSize": 12,
                      "fontWeight": "normal",
                    },
                  ]
                }
                uppercase={false}
              >
                4:00 PM
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
